packages/repository/src/relations/has-many/has-many-through.helpers.ts

```diff
@@ -109,6 +119,45 @@ export function createThroughConstraint<Through extends Entity, ForeignKeyType>(
   const constraint: any = {[sourceFkName]: fkValue};
   return constraint;
 }
+/**
+ * Creates constraint used to create the through model
+ *
+ * @param relationMeta - resolved hasManyThrough metadata
+ * @param targetInstance instance of target entity used to constrain through
+ * @internal
+ *
+ * @example
+ * ```ts
+ * const resolvedMetadata = {
+ *  // .. other props
+ *  keyFrom: 'id',
+ *  keyTo: 'id',
+ *  through: {
+ *    model: () => CategoryProductLink,
+ *    keyFrom: 'categoryId',
+ *    keyTo: 'productId',
+ *  },
+ * };
+ * createThroughConstraint(resolvedMetadata, {id: 3, name: 'a product'});
```

Comment:
The additional query field reminds me sth(sorry don't remember where is the original chat :p) : 
 
We want to distinguish the query on through model(CategoryProductLink) and target model(Product), how do we achieve that? 

For example, `{name: 'a product'}` seems for target model, and `createThroughConstraint` applies for through model, and that's why `name` is ignored here. But:

- where is the place that processes the target model's filter?
- how to specify the through model's filter?

(I started from the helper function, might missing the code/docs if they are created other places.)

Reply:
@jannyHou I think these helpers are just for handling the relation metadata. And filters are handled in the hasManyThrough repository. Here is an example in the original design: [find method](https://github.com/strongloop/loopback-next/pull/2359/files#diff-bd2fafeb3bd8f873174e06856a1d7471R124-R141). They won't be included in this PR. But I'd love to discuss it: 

So far from those two community PRs(#2359 , #4438 ), I only see the target  model is capable of applying filters. Not sure if we need it for through models ( I don't see the reason to allow filtering through models tho). Do we need it?